clients/spark/core/src/test/scala/io/treeverse/clients/ApiClientSpec.scala

```diff
@@ -32,17 +38,136 @@ class ApiClientSpec extends AnyFunSpec with Matchers {
     }
 
     describe("Azure") {
-      val translate = (u: String) => ApiClient.translateURI(new URI(u), StorageUtils.StorageTypeAzure).toString
+      val translate =
+        (u: String) => ApiClient.translateURI(new URI(u), StorageUtils.StorageTypeAzure).toString
       it("should translate any protocol to abfs") {
-        translate("https://account.example.net/container/path/to/blob") should be("abfs://container@account.dfs.core.windows.net/path/to/blob")
-        translate("ex://account.example.net/container/path/to/blob") should be("abfs://container@account.dfs.core.windows.net/path/to/blob")
+        translate("https://account.example.net/container/path/to/blob") should be(
+          "abfs://container@account.dfs.core.windows.net/path/to/blob"
+        )
+        translate("ex://account.example.net/container/path/to/blob") should be(
+          "abfs://container@account.dfs.core.windows.net/path/to/blob"
+        )
       }
 
       it("should handle empty paths") {
         // TODO(lynn): Trailing slashes might well be incorrect here.
-        translate("https://account.example.net/container/") should be("abfs://container@account.dfs.core.windows.net/")
-        translate("https://account.example.net/container") should be("abfs://container@account.dfs.core.windows.net/")
+        translate("https://account.example.net/container/") should be(
+          "abfs://container@account.dfs.core.windows.net/"
+        )
+        translate("https://account.example.net/container") should be(
+          "abfs://container@account.dfs.core.windows.net/"
+        )
       }
     }
   }
 }
+
+class RequestRetryWrapperSpec extends AnyFunSpec with Matchers with BeforeAndAfterEach {
+  val SomeValue = "some value"
+  val DefaultMaxNumRetries = 5
+  val DefaultMaxNumAttempts = 6
+
+  var retryWrapper = spy(new RequestRetryWrapper)
+  var dummyMethodInvoker = spy(new DummyMethodInvoker)
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    retryWrapper = spy(new RequestRetryWrapper)
+    assert(retryWrapper.numRetries == DefaultMaxNumRetries)
+    dummyMethodInvoker = spy(new DummyMethodInvoker)
+  }
+
+  class DummyMethodInvoker {
+    @throws[Exception]
```

Comment:
I don't understand why we need this?  AFAICT `someMethod` does _not_ ever throw `Exception`.